content/docs/rfcs/17/README.md

```diff
@@ -2,127 +2,136 @@
 slug: 17
 title: 17/WAKU-RLN-RELAY
 name: Waku v2 RLN Relay
-status: raw
+status: draft
 tags: waku-core
 editor: Sanaz Taheri <sanaz@status.im>
 ---
 
-The current specification embodies the details of the spam-protected version of `relay` protocol empowered by Rate Limiting Nullifiers (RLN). 
-<!-- More details on RLN can be found in [this spec]() (TODO: to link the spec). -->
+The `17/WAKU-RLN-RELAY` protocol is an extension of `11/WAKU-RELAY` which additionally provides spam protection using [Rate Limiting Nullifiers (RLN)](/spec/32). 
 
-The security objective is to control the number of PubSub messages that each peer can publish per epoch where epoch is a system design parameter, regardless of the published topic.
+The security objective is to contain spam activity in a GossipSub network by enforcing a global messaging rate to all the peers.
+Peers that violate the messaging rate are considered spammers and their message is considered spam.
+Spammers are also financially punished and removed from the system. 
 
 
 **Protocol identifier***: `/vac/waku/waku-rln-relay/2.0.0-alpha1`
 
 # Motivation
 
-In open p2p messaging networks, one big problem is spam-resistance. 
-Existing solutions, such as Whisper’s proof of work, are insufficient, especially for heterogeneous nodes. 
+In open and anonymous p2p messaging networks, one big problem is spam resistance. 
+Existing solutions, such as Whisper’s proof of work are computationally expensive hence not suitable for resource-limited nodes. 
 Other reputation-based approaches might not be desirable, due to issues around arbitrary exclusion and privacy.
 
-We augment the `relay` protocol with a novel, light, and effective spam prevention mechanism which also suits the resource-constrained nodes.
-
-<!-- TODO: Fill in more -->
+We augment the [`11/WAKU2-RELAY`](/spec/11) protocol with a novel construct of [RLN](/spec/32) to enable an efficient economic spam prevention mechanism that can be run in resource-constrained environments.
 
 
 # Flow
+The messaging rate is set to 1 per `epoch`, 
+where `epoch` indicates the number of `T` seconds elapsed since the Unix epoch event.
+The value of `T` is application-dependent.
+See section [Recommended System Parameters](#recommended-system-parameters) for some recommended ways to set a sensible `T` value.
+Peers subscribed to a spam-protected `pubsubTopic` are only allowed to send one message per `epoch`
```

Comment:
Is one message hardcoded or a variable? Either case should make it explicit

Reply:
Messaging rate is always 1 per some unit of time which is defined by `T`. There are a few points about higher messaging rates i.e., >1:
- The current rln lib does not have support for a higher messaging rate (higher degree polynomials)
- In the other library developed by Blagoj, the support for a higher messaging rate is excluded as the same effect can be captured by taking the epoch duration. This means probably messaging rate is going to be always 1.
-  Changing the messaging rate to higher than 1 will affect the slashing process described in the spec (for which there is no clear definition of internal nullified either), so I'd say let's keep it simple and stick to the messaging rate of 1.

Reply:
That makes sense! In that case we can make it explicitly that it is hardcoded to message per epoch per definition.

I'm still not clear the use of `T` here, seems like this could be clarified. See https://github.com/vacp2p/rfc/pull/484#discussion_r803289130

Reply:
> That makes sense! In that case we can make it explicitly that it is hardcoded to message per epoch per definition.

Sure, I'll update it

> I'm still not clear the use of T here, seems like this could be clarified. See #484 (comment)

Please see this comment of mine https://github.com/vacp2p/rfc/pull/484/files#r804929807